### PR TITLE
Reporter implementation

### PR DESCRIPTION
**Pull request by orelmisan**
Add the Reporter implementation, based on the VM latency checkup's reporter [1].
Add the ability to report checkup status to the user.

[1] https://github.com/kiagnose/kiagnose/tree/95d8c7995fabbb7be11f7cde0eca57dc01e222bb/checkups/kubevirt-vm-latency/vmlatency/internal/reporter

Manually tested on an OpenShift 4.11 cluster.

~~This PR depends on PR #11.~~

### CODE REVIEW

File: pkg/internal/reporter/reporter.go [hunk 1]
```diff
@@ -19,15 +19,53 @@
...
+	return &Reporter{*r}
 }
 
 func (r *Reporter) Report(checkupStatus status.Status) error {
-	return nil
+	if !r.HasData() {
+		return r.Reporter.Report(checkupStatus.Status)
+	}
+
+	// TODO: Update the base reporter to drop the `Succeeded` field.
```
Comment (RamLavi) [inline pkg/internal/reporter/reporter.go]: do you mean the kiagnose reporter?

Comment (orelmisan) [reply pkg/internal/reporter/reporter.go]: Yes, this is a comment originally made by @EdDev.
He wished to remove the status.Succeeded field from kiagnose, and calculate it from the errors.

**Comment by orelmisan** [comment]
Rebase

File: pkg/internal/reporter/reporter.go (hunk 1 above)
Comment (RamLavi) [reply pkg/internal/reporter/reporter.go]: well, if you are TODOing a task to perform on the kiagnose/kiagnose repo - then I would suggest not..
Why not just correct it here?

**Comment by orelmisan** [comment]
[Change](https://github.com/kiagnose/kubevirt-rt-checkup/compare/bc072db7ebb645120f05e6e8749bc7bb74098e9e..c9fd7121277c42405a61710978f466149f8a0d9e): Added an uncommited change to go.mod.

File: pkg/internal/reporter/reporter.go (hunk 1 above)
Comment (orelmisan) [reply pkg/internal/reporter/reporter.go]: The TODO talked about a change in the kiagnose library.
We need to fix it there.

Comment (RamLavi) [reply pkg/internal/reporter/reporter.go]: ok, then why do we mention it here? this is a different repo.

Comment (orelmisan) [reply pkg/internal/reporter/reporter.go]: The comment was taken from the VM latency checkup.
I will remove the comment from here.
When we will fix kiagnose, this line `checkupStatus.Succeeded = len(checkupStatus.FailureReason) == 0` will be removed.

**Comment by RamLavi** [comment]
> [Change](https://github.com/kiagnose/kubevirt-rt-checkup/compare/bc072db7ebb645120f05e6e8749bc7bb74098e9e..c9fd7121277c42405a61710978f466149f8a0d9e): Added an uncommited change to go.mod.

yay for the go tidy check!

**Review by RamLavi — COMMENTED** [review]
@orelmisan do you mean to only add the code regarding Oslat?

**Comment by orelmisan** [comment]
> @orelmisan do you mean to only add the code regarding Oslat?

Yes, for now.
The cyclictest will come later.

The config parser also takes just the oslat part.

**Review by RamLavi — APPROVED** [review]
/lgtm

**Pull request merged**